src/tsk_fs_file.rs

```diff
@@ -134,6 +136,54 @@ impl<'fs> TskFsFile<'fs> {
         let meta = unsafe { (*self.tsk_fs_file_ptr).meta };
         unsafe{*meta}.type_ & tsk::TSK_FS_META_TYPE_ENUM_TSK_FS_META_TYPE_DIR > 0
     }
+
+    /// Get the TskFsMeta for this TskFsFile
+    pub fn get_meta(&self) -> Result<TskFsMeta, TskError> {
+        TskFsMeta::from_ptr(unsafe{(*self.tsk_fs_file_ptr).meta})
+    }
+
+    /// Get an iterator that reads the file in chunks of buf_size.
+    pub fn read_iter(&'fs self, buf_size: u64) -> TskFsFileIterator<'fs> {
+        let file_size = self.get_meta().unwrap().size();
+        TskFsFileIterator{
+            tsk_fs_file: self,
+            offset:0,
+            file_size: file_size as u64,
+            buf_size
+        }
+    }
+
+    /// Reads the exact buffer size from the file
+    pub fn read_exact(&self,offset:i64, buff: &mut [u8]) -> Result<u64,TskError>{
+        let bytes_read = unsafe{tsk::tsk_fs_file_read(
+            self.into(),
+            offset,
+            buff.as_mut_ptr() as *mut i8,
+            buff.len() as u64,
+            0
+        )};
+        match bytes_read {
+            -1 => {
+                // Get a ptr to the error msg
+                let error_msg_ptr = unsafe { tsk::tsk_error_get() };
+                // Get the error message from the string
+                let error_msg = unsafe { CStr::from_ptr(error_msg_ptr) }.to_string_lossy();
+                return Err(TskError::tsk_fs_file_error(
+                    format!("Error on read_exact() : {}", error_msg)
+                ));
+            }
+            _ => return Ok(bytes_read as u64),
+        };
+    }
+
+    /// Reads the file in chuncks to the specified path
+    pub fn read_to(&self,path:&str) -> Result<bool,TskError>{
```

Comment:
I would rather have Read/Seek traits implemented for TskFsFile. Because of that, it would be better for a tool that uses this library to determine how it wants to read and write to a file. I don't think this should be on the library level.

Reply:
What about making the `TskFsFileHandler` accept `TskFsAttrType` & `TskFsFileReadFlag` and implementing `Read` & `Seek` on `TskFsFile` which only accepts `TskFsFileReadFlag` and read the attribute `NTFS_DATA` ?